cirq/circuits/frozen_circuit.py

```diff
@@ -85,13 +89,19 @@ def moments(self) -> Sequence['cirq.Moment']:
     def device(self) -> devices.Device:
         return self._device
 
+    @property
+    def name(self) -> Optional[str]:
+        return self._name
+
     def __hash__(self):
-        return hash((self.moments, self.device))
+        return hash((self.moments, self.device, self.name))
+
+    def _serialization_name_(self):
```

Comment:
It doesn't look like this method is used outside of tests. Since it's "private", consider removing it (or using it).

Reply:
This is a "magic method"; defining it here is what enables concise serialization of `FrozenCircuit`s in the JSON serialization protocol.

Reply:
Is it possible to reference the overridden method so that we can easily access its documentation? Should this class also extend [SerializableByKey](https://github.com/quantumlib/Cirq/blob/0677d6082bad22289c8438596dd7bf815c21e1c2/cirq/protocols/json_serialization.py#L332) to be explicit about this support?

Reply:
I'm not fundamentally opposed to adding a reference here, but in general magic methods in Cirq don't reference their protocols. Instead, there's an implicit assumption that magic method usage is defined in the relevant protocol; this is supported by the [protocols doc](https://github.com/quantumlib/Cirq/blob/master/docs/protocols.ipynb).